src/main/java/net/dv8tion/jda/internal/entities/PermissionOverrideImpl.java

```diff
@@ -103,12 +105,21 @@ public JDA getJDA()
         return api;
     }
 
+    @Nonnull
+    @Override
+    public IPermissionHolder getPermissionHolder()
```

Comment:
You mark this as `Nonnull` but both methods used for resolution may return null. This is an error, the correct annotation to use is `@Nullable`.

Reply:
I thought it is Nonnull because the PermissionOverride is for a Member or a Role and those methods return null if the override is for the other entity. So checking both should cover all cases making it Nonnull?

Reply:
No. It's nullable.

Reply:
Will mark it as Nullable. For a better understanding, could you give me an example in which case it could return null? Or is it nullable just by definition (all methods used for it are nullable)?

Reply:
If we change `getPermissionHolder` to not call `getRole/getMember` (which arguably we should), then this _can_ be marked `@Notnull` as the underlying implementation of the PermissionOverride will be providing the explicit entity

Reply:
Ah, actually no. I didn't realize we weren't storing an actual reference to a `IPermissionHolder` object or soft-reference. You'll note that we're only storing the id here: https://github.com/DV8FromTheWorld/JDA/blob/ed2b57b38d357de1df18773178997b88f2a7ac6b/src/main/java/net/dv8tion/jda/internal/entities/PermissionOverrideImpl.java#L59
This means that a lookup of `getRoleById` or `getMemberById` can return null due to timing or cache invalidate.

This is a question here as to whether we should drop using the `id` only and instead have a reference to the IPermissionHolder object instead as there is actual use for it here.

Reply:
Didn't thought about the cache invalidation, good point!

The IPermissionHolder was stored as a class variable until 2 weeks ago. And because I didn't want to store it again, I made it like this.
[Link to the commit](https://github.com/DV8FromTheWorld/JDA/commit/4426e2e2e8eda2faff049f4503dfe0cd0c478f32#diff-3acd3d97f0f475cc7d2d1fadb219b3f0)

Reply:
I have plans to make these available even if the member is currently not cached. So returning null is in all situations the behavior I want to convey as correct here.